app/classifier/tasks/dropdown/index.cjsx

```diff
@@ -82,23 +82,21 @@ module.exports = React.createClass
         not select.required or (annotation.value[i]?.value? and annotation.value[i]?.value isnt "")
 
   getInitialState: ->
-    selectedOptions: []
+    optionsKeys: {}
 
   componentWillMount: ->
     @menus = []
     @syncAnnotations()
 
-  componentDidMount: ->
-    annotationValues = @props.annotation.value
-    unless annotationValues.length
-      @props.annotation.value = @props.task.selects.map -> {value: null, option: false}
+  componentWillReceiveProps: (nextProps) ->
+    if nextProps.task isnt @props.task
+      @syncAnnotations()
```

Comment:
This will call `syncAnnotations` before `props.task` has updated. Just a heads up because there's a comment on that method that warns not to do this.